packages/client/src/components/charts/RegRatesLineChart.tsx

```diff
@@ -131,24 +113,49 @@ function LegendDot(props: React.HTMLAttributes<SVGElement>) {
   )
 }
 
-class RegRatesLineChartComponent extends React.Component<IProps, IState> {
-  constructor(props: IProps) {
-    super(props)
-    this.state = {
-      ...this.getStatePropertiesForChart(),
-      ...this.getLatestData()
-    }
-  }
+function RegRatesLineChartComponent (props: IProps) {
 
-  getStatePropertiesForChart = () => {
-    if (window.innerWidth > this.props.theme.grid.breakpoints.md) {
-      return this.getStatePropertiesForLargeWindowChart()
+  const getStatePropertiesForChart = () => {
+    if (window.innerWidth > props.theme.grid.breakpoints.md) {
+      return getStatePropertiesForLargeWindowChart();
     } else {
-      return this.getStatePropertiesForSmallWindowChart()
+      return getStatePropertiesForSmallWindowChart();
     }
-  }
+  };
+
+  const getLatestData = () => {
+    const { data, theme } = props;
+    const latestData = data && data[data.length - 1];
+    return {
+      activeLabel: (latestData && latestData.label) || '',
+      activeRegisteredInTargetDays: {
+        value: (latestData && latestData.registeredInTargetDays) || 0,
+        stroke: theme.colors.teal
+      },
+      activeTotalRegistered: {
+        value: (latestData && latestData.totalRegistered) || 0,
+        stroke: theme.colors.tealLight
+      },
+      activeTotalEstimate: {
+        value: (latestData && latestData.totalEstimate) || 0,
+        stroke: theme.colors.grey300
+      }
+    };
+  };
+
+  const [state, setState] = useState({
+    ...getStatePropertiesForChart(),
+    ...getLatestData()
+  });
 
-  getStatePropertiesForSmallWindowChart = () => {
+  useEffect(() => {
+    window.addEventListener('resize', recordWindowWidth);
+    return () => {
+      window.removeEventListener('resize', recordWindowWidth);
+    };
+  }, []);
```

Comment:
We can directly use the `useWindowSize` hook right?

Reply:
In this case, I think useWindowSize can't be used because it handles the state;
{
    width: window.innerWidth,
    height: window.innerHeight
  }

yet we are dealing with the state;

{
  legendMarginTop: number
  legendMarginLeft: number
  chartTop: number
  chartRight: number
  chartBottom: number
  chartLeft: number
  maximizeXAxisInterval?: boolean
  legendLayout: LegendProps['layout']
  activeLabel: string
  activeRegisteredInTargetDays: IActiveState
  activeTotalRegistered: IActiveState
  activeTotalEstimate: IActiveState
}

Reply:
We can separate those 2 states out right? They should be independent of each other

Reply:
Okay I get what you are saying. Separating those would be a bit more work. we can leave it as it is now